polling/frequent/activities.go

```diff
@@ -0,0 +1,32 @@
+package frequent
+
+import (
+	"context"
+	"github.com/temporalio/samples-go/polling"
+	"go.temporal.io/sdk/activity"
+	"time"
+)
+
+// @@@SNIPSTART samples-go-polling-frequent-activities
+
+type PollingActivities struct {
+	TestService  *polling.TestService
+	PollInterval time.Duration
+}
+
+// DoPoll Activity.
+// In this activity polling is implemented within the activity itself and not the workflow,
+// using the heartbeat mechanism to keep the activity alive
+func (a *PollingActivities) DoPoll(ctx context.Context) (string, error) {
+	for {
+		res, err := a.TestService.GetServiceResult()
+		if err == nil {
+			return res, err
+		}
+		activity.RecordHeartbeat(ctx, "no response")
```

Comment:
I would not provide details in this case as they are intended to record the activity progress. In the future, we might skip a DB write if the details are empty.

Reply:
Makes sense, should I also update the Java example here? https://github.com/temporalio/samples-java/blob/main/src/main/java/io/temporal/samples/polling/frequent/FrequentPollingActivityImpl.java#L51

Reply:
https://github.com/temporalio/samples-java/pull/412